### PR TITLE
Prototype trusted types implementation

### PR DESCRIPTION
**Pull request by lukewarlow**
## Description

PR on fork so it doesn't add noise to main repo.

TLDR this demonstrates roughly what changes would be needed to make HTMX [trusted types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API) compliant.

Also the policies currently just pass through the string, it would need to be thought out whether any of these sinks are being passed untrusted data from the front end that should maybe be passed through an actual policy with some sanitising.

Corresponding issue:

## Testing
*Please explain how you tested this change manually, and, if applicable, what new tests you added. If
you're making a change to just the website, you can omit this section.*

## Checklist

* [ ] I have read the contribution guidelines
* [ ] I have targeted this PR against the correct branch (`master` for website changes, `dev` for
  source changes)
* [ ] This is either a bugfix, a documentation update, or a new feature that has been explicitly
  approved via an issue
* [ ] I ran the test suite locally (`npm run test`) and verified that it succeeded

### CODE REVIEW

File: src/htmx.js
Comment (lukewarlow) [inline src/htmx.js:1]: On line 1861 there's the below code, if target can ever be a script element this will cause a TT violation. So we'd need to do an if target is HTMLScriptElement then use the policy.

```js
if (swapSpec.swapStyle === 'textContent') {
      target.textContent = content
```

File: src/htmx.js [hunk 1]
```diff
@@ -1,6 +1,37 @@
 var htmx = (function() {
   'use strict'
 
+  if (!trustedTypes) {
+    // @ts-ignore
+    trustedTypes = {
```
Comment (lukewarlow) [inline src/htmx.js]: We should maybe assign this to a local variable and likewise with the actual impl if it exists, and then use that, so that htmx isn't polluting the window with a fake polyfill.

File: src/htmx.js [hunk 2]
```diff
@@ -2050,7 +2081,7 @@ var htmx = (function() {
             conditionalSource += ')})'
             try {
               const conditionFunction = maybeEval(elt, function() {
-                return Function(conditionalSource)()
+                return Function(htmxPolicy.createScript(conditionalSource))()
```
Comment (lukewarlow) [inline src/htmx.js:2167]: This won't actually work in Chromium atm due to a bug with Trusted Types implementation

You'd have to use eval instead. See https://github.com/angular/angular/blob/3bfba7ec1530d451f53e5d85516a33400550bbb3/packages/core/src/util/security/trusted_types.ts#L119

File: src/htmx.js [hunk 3]
```diff
@@ -2741,7 +2772,9 @@ var htmx = (function() {
           return
         }
         if (!func) {
-          func = new Function('event', code)
+          // TODO: This won't actually work in Chromium atm due to a bug with Trusted Types implementation
+          // You'd have to use eval instead. See https://github.com/angular/angular/blob/3bfba7ec1530d451f53e5d85516a33400550bbb3/packages/core/src/util/security/trusted_types.ts#L119
+          func = new Function('event', htmxDemoPolicy.createScript(code))
```
Comment (lukewarlow) [inline src/htmx.js]: This won't actually work in Chromium atm due to a bug with Trusted Types implementation

You'd have to use eval instead. See https://github.com/angular/angular/blob/3bfba7ec1530d451f53e5d85516a33400550bbb3/packages/core/src/util/security/trusted_types.ts#L119

File: src/htmx.js [hunk 4]
```diff
@@ -3729,7 +3762,7 @@ var htmx = (function() {
       }
       let varsValues
       if (evaluateValue) {
-        varsValues = maybeEval(elt, function() { return Function('return (' + str + ')')() }, {})
+        varsValues = maybeEval(elt, function() { return Function(htmxPolicy.createScript('return (' + str + ')'))() }, {})
```
Comment (lukewarlow) [inline src/htmx.js:3876]: This won't actually work in Chromium atm due to a bug with Trusted Types implementation

You'd have to use eval instead. See https://github.com/angular/angular/blob/3bfba7ec1530d451f53e5d85516a33400550bbb3/packages/core/src/util/security/trusted_types.ts#L119

File: src/htmx.js [hunk 5]
```diff
@@ -4540,7 +4576,7 @@ var htmx = (function() {
     if (title) {
       const titleElt = find('title')
       if (titleElt) {
-        titleElt.innerHTML = title
+        titleElt.textContent = title
```
Comment (lukewarlow) [inline src/htmx.js:4705]: Title elements can't have HTML children so innerHTML here is pointless.
